haystack/nodes/ranker/recentness_ranker.py

```diff
@@ -0,0 +1,147 @@
+
+
+class RecentnessRanker(BaseRanker):
+    outgoing_edges = 1
+
+    def __init__(
+        self,
+        date_identifier: str,
+        weight: float = 0.5,
+        top_k: Optional[int] = None,
+        method: str = "reciprocal_rank_fusion",
+    ):
+        """
+        This Node is used to rerank retrieved documents based on their age. Newer documents will rank higher.
+        The importance of recentness is parametrized through the weight parameter.
+
+        :param date_identifier: Identifier pointing to the date field in the metadata.
+                This is a required parameter, since we need dates for sorting.
+        :param weight: in range [0,1].
+                0 disables sorting by age.
+                0.5 content and age have the same impact
+                1 means sorting only by age, most recent comes first.
+        :param top_k: (optional) How many documents to return.
+                It can make sense to have large top-k values from the initial retrievers and filter docs down in the
+                RecentnessRanker with this top_k parameter.
+        :param method: Method used to combine retriever and recentness. Possible values are 'reciprocal_rank_fusion' (default) and 'score'
+                Make sure to use 'score' method only with retrievers/rankers that give back OK score in range [0,1]
+        """
+
+        super().__init__()
+        self.date_identifier = date_identifier
+        self.weight = weight
+        self.top_k = top_k
+        self.method = method
+
+    # pylint: disable=arguments-differ
+    def run(self, query: str, documents: List[Document], top_k: Optional[int] = None) -> Tuple[Dict, str]:  # type: ignore
+        # sort documents based on age, newest comes first
+        try:
+            sorted_by_date = sorted(documents, reverse=True, key=lambda x: parse(x.meta[self.date_identifier]))
+        except KeyError:
+            logger.error(
+                """
+                Param <date_identifier> seems wrong.\n
+                You supplied: '%s'.\n
+                Document[0] contains metadata with keys: %s.\n
+                Continuing without sorting by date.
+                """,
+                self.date_identifier,
+                ",".join(list(documents[0].meta.keys())),
+            )
+            output = {"documents": documents}
+            return output, "output_1"
+        except ParserError:
+            logger.error(
+                """
+                Could not parse date information for dates: %s\n
+                Continuing without sorting by date.
+                """,
+                " - ".join([x.meta.get(self.date_identifier, "identifier wrong") for x in documents]),
+            )
+            output = {"documents": documents}
+            return output, "output_1"
+
+        # merge scores for both documents sorted by content and date.
+        # We use reciprocal rank fusion (RRF) to have the same range of scores for both lists.
```

Comment:
> \# We use reciprocal rank fusion (RRF) to have the same range of scores for both lists.

This is only true if RRF is chosen as the method. Perhaps something like the following?

> \# If method is set to `'reciprocal_rank_fusion'`, then that is used to combine previous ranking with recency ranking.
> \# If method is set to `'score'`, then documents will be assigned a recency score in [0,1] and will be re-ranked based on the weighted average of their recency score and their pre-existing score.

Of course I defer to those with more expertise on technical writing here :)